zerver/lib/test_classes.py

```diff
@@ -1266,7 +1266,10 @@ def tornado_redirected_to_list(self, lst: List[Mapping[str, Any]]) -> Iterator[N
         # Some code might call process_notification using keyword arguments,
         # so mypy doesn't allow assigning lst.append to process_notification
         # So explicitly change parameter name to 'notice' to work around this problem
-        yield
+
+        with self.captureOnCommitCallbacks(execute=True):
+            yield
```

Comment:
I'd put a polished version commit message explanation in a nice block comment here, since it's important to understanding what's happening here in our test infrastructure.  

(Also the existing comments for this function seem misplaced; I'd expect them to have been above the hacky functions).

Reply:
As a sidenote, one of the most important parts of having a readable codebase is asking the question "Where does this explanation belong?" -- a comment, a commit message, a function name, documentation in ReadTheDocs with a link are all valid answers, and choosing the right one is a big part of making a codebase happy.

Reply:
Done.